lib/sycamore/sycamore/llms/openai.py

```diff
@@ -352,6 +395,63 @@ def _generate_using_openai(self, prompt_kwargs, llm_kwargs) -> str:
         completion = self._client.chat.completions.create(model=self._model_name, messages=messages, **kwargs)
         return completion.choices[0].message.content
 
+    async def generate_future(self, *, prompt_kwargs: dict, llm_kwargs: Optional[dict] = None) -> Awaitable[str]:
+        cache_key = None
+        if self._cache:
+            cache_key = self._get_cache_key(prompt_kwargs, llm_kwargs)
+            hit = self._cache.get(cache_key)
```

Comment:
```
assert hit.get("prompt_kwargs") == prompt_kwargs and hit.get("llm_kwargs") == llm_kwargs and hit.get("model_name") == self.model.name
```
No point in continuing if the cache is somehow broken and much shorter.

Reply:
I'd move this into a function and call it. Then generate can share the same code. So:
```
   (key, ret) = self._cache_get(prompt_kwargs, llm_kwargs)
   if ret is not None:
       return ret
  ...
   self._cache_set(key, result)
```

Then you can fix a caching bug.
```
   if llm_kwargs.get("temperature", 0) != 0:
       return (None, None)
```

Reply:
changed -- implementation also does not store if the temperature is non zero.